src/xmipp/libraries/data/image_operate.cpp

```diff
@@ -531,7 +531,8 @@ void ProgOperate::readParams()
             {
                 if (mdInSize != md2.size())
                     REPORT_ERROR(ERR_MD, "Both metadatas operands should be of same size.");
-                md2Iterator.init(md2);
+                md2IdIterator = std::unique_ptr<MetaDataVec::id_iterator>(new MetaDataVec::id_iterator(md2.ids().begin()));
```

Comment:
This looks over complicated. Do we need a pointer to iterator?

Reply:
Yes, because at the creation of `ProgOperate` class we don't know how to set default value to iterator. We could use `std::optional.`.

Reply:
Code siplified using `memoryUtils::makeUnique`.